client.go

```diff
@@ -219,18 +219,15 @@ func (co *Conn) ReadMsgHeader(hdr *Header) ([]byte, error) {
 		n   int
 		err error
 	)
-
-	switch t := co.Conn.(type) {
+	switch co.Conn.(type) {
 	case *net.TCPConn, *tls.Conn:
-		r := t.(io.Reader)
-
-		// First two bytes specify the length of the entire message.
-		l, err := tcpMsgLen(r)
-		if err != nil {
+		var length uint16
+		if err := binary.Read(co.Conn, binary.BigEndian, &length); err != nil {
```

Comment:
is there any way to test this for short reads, i.e. see comments in tcpMsgLen where you only get 1 byte on this read?

Reply:
`binary.Read` uses `io.ReadFull` internally which handles this case. That being said, I could try and build a test using `testing/iotest`.